### ISSUE TITLE
Android, Freeze after Add wallet (spinner never finishes)   

### ISSUE THREAD

**Issue by sgornick** [BlueWallet/BlueWallet#2231]
After installing the app, creating the wallet appears to complete OK.  But then swiping down to view transactions causes a spinner to spin, and it never completes (at least, not within 5 minutes). 

Steps to replicate: 

1. Install BlueWallet (or Force Stop, and Clear Data for app)
2. On startup, wallets appears.  Press Add now button
3. In add wallet screen, press to select Bitcoin (blue), and then press Create button
4. In Your wallet is created ... screen, press Ok, I wrote this down! button
5. In wallets screen, Wallet with 0 BTC appears.  Swipe down to trigger refresh of transaction history    
6. Spinner will spin, for at least five minutes, with no error, and no completion.  The UI is still responding, I can swipe left to navigate to the Add a wallet panel, but that's about it (e.g., the Add now button does not respond).  I can even try doing a Force Stop, and then after launching BlueWallet again it will just sit there with the spinner. 

Expected behavior:

Spinner to complete in a reasonable amount of time, especially with the app being newly installed and this being the only wallet. 

The device I'm experiencing this on is an Alcatel 5041c:
http://specdevice.com/showspec.php?id=6b4a-c88f-ffff-ffffd91d6a7a

It is a fairly low-end device which currently has no voice or mobile data (SIM card installed, but no service and the Network -> Mobile network -> Domestic data slider is disabled), only Wi-Fi.  Not sure if this is a contributing factor, but wanted to mention it.   I have previously had used BlueWallet on this device when it had service, but have since reset to Factory (for use as a dedicated bitcoin wallet) and this is the first time trying to create a bitcoin wallet in BlueWallet since doing that.

One other relevant detail, ... if I first (after Force Stop and Clear Data) import a Lightning network wallet (scanning an LNDHub), then that LN wallet works fine -- I can see LN transactions and that's all fine.  But then adding a Bitcoin wallet gives the exact same results as in step 6, above.  

Android version: 8.1.0,  Android security patch level: October 5, 2020, Kernel version: 4.4.95
BlueWallet ver 5.6.6

**Comment by sgornick** [comment]
Ok, I just noticed the Self Test.

Running that gives: 

`Error: Could not fetch HD Bech32 balance`

**Comment by marcosrdz** [comment]
Go to Settings > electrum server.

Does it display "Connected"?

**Comment by sgornick** [comment]
> Does it display "Connected"?

Yes.  Connected,  electrum1.bluewallet.io:443

But About -> Run self test gives:

`Error: Could not fetch HD Bech32 balance`

**Comment by Overtorment** [comment]
is the problem still there?

can you try tris: on electrum settings screen
![image](https://user-images.githubusercontent.com/1913337/100759791-f02f6800-33e8-11eb-8b59-8ad42dfb6a9a.png)


tap the text with electrum address. it should ping the server and throw an alert with response:

![image](https://user-images.githubusercontent.com/1913337/100759888-0e956380-33e9-11eb-99fa-98564023d5e5.png)

**Comment by Overtorment** [comment]
my guess is that something is weird with your wifi configuration. 
maybe, some additional firewall that doesnt like the fact that TLS connection with electrum on port 443 has self-signed certificate..?

can you try other wifi networks?

**Comment by sgornick** [comment]
> is the problem still there?

Yes.

> can you try tris: on electrum settings screen
![image](https://user-images.githubusercontent.com/24884/100999468-6f748700-3522-11eb-8a10-36a018ec4506.png)

![image](https://user-images.githubusercontent.com/24884/100999473-70a5b400-3522-11eb-8e93-a88276014a3c.png)

Again, this is with mobile device with no service on the SIM card that is installed.  I also just now tested without the SIM card, just in case a SIM card installed but out of service was some contributing factor, but I get the same results regardless of SIM card installed or not. 

> maybe, some additional firewall that doesnt like the fact that TLS connection with electrum on port 443 has self-signed certificate..?

Is there some way of troubleshooting that?  Or a debug version that will show the error?

I know nothing about certificates, but I see that SSL scanner by Qualys SSL Labs stops scanning and reports: "Certificate name mismatch". 

https://www.ssllabs.com/ssltest/analyze.html?d=electrum3.bluewallet.io

Ignoring that and proceeding does yield a handshake simulation (for my version of Android, 8.1, a green color for TLS 1.3 --- whatever that means. But just sharing this here in case there might actually be something here that isn't looking right.  I would assume that any TLS / certificate issue would affect everyone (at least, everyone on Android 8.1), and not just access from my one device).  

>  my guess is that something is weird with your wifi configuration.
>
> maybe, some additional firewall 

The wi-fi is a Linksys E2500, with all factory settings with the exception of having the [router name]-guest SSID enabled.  I get the same thing whether I am using the [router name]-guest SSID as well as just the [router name] SSID, which would tell me there is nothing that this Linksys router would be doing any different from any other router.  On the WAN side of that is a consumer grade Comcast (Xfinity) cable modem.  I can't imagine there's any firewalling done with that.

I do keep forgetting to try with another wi-fi though, when I've been out and about.  I will do a test on another Wi-Fi connection and reply with the results.  If it isn't just related to this one wi-fi and cablemodem connection, then maybe there's some other way for me to do some further troubleshooting.

**Comment by ncoelho** [comment]
Getting back to the original problem. How many transactions does this
wallet have?

**Comment by Overtorment** [comment]
ok, the fact that you see this alert means that connection is fine (data inside alert came startight from electrum) and the problem is elsewhere.

lets try following:

1) can you import single address as a wallet? preferably with few transactions, so that it imports faster. just import any address (like `bc1qwte78pw877jas80xgyyffgynv8v4wpggu82x3q`) and it whould import as watch-only. try pull-to-refresh and see if transactions will get loaded.
2) try reproducing again this endless spinner couple of times, then get your public IP address and pls share it privately to i@bluewallet.io, I will look it up on our crash reporting service

**Comment by sgornick** [comment]
ncoelho wrote:
> Getting back to the original problem. How many transactions does this
wallet have?

This is occurring with a fresh install, and thus newly created wallet (which, of course, would have no transactions).

If you are referring to an issue I had months ago (and sent an e-mail to support), with importing a seed, this issue reported here is occurring on the same device, but it has since been factory reset since the earlier (unresolved) issue, and I haven't tried to import that seed since.  

Overtorment wrote:
> import any address (like bc1qwte78pw877jas80xgyyffgynv8v4wpggu82x3q) and it whould import as watch-only. t

With starting clean (i.e., no wallet existing before importing this), the app displays the balance for the Imported Watch-only wallet immediately.  And swiping down refreshes promptly.  As soon as I create a wallet, and swipe down, the never-ending spinner occurs. 

Is this simply a corner case issue that nobody has tested?  Where if a wallet is created, the check for the balance fails (as there is no activity to return)?

**Comment by Overtorment** [comment]
> Is this simply a corner case issue that nobody has tested? Where if a wallet is created, the check for the balance fails (as there is no activity to return)?

the fact that that your selftest shows an error is very weird. this is definately edge case, because for most of us (and thousands of others) everything works fine, and selftest is always checked prior to release via automated tests.

lets try to find your case on crash reporting service (I would need your IP for that on my email, thanks)

**Comment by Overtorment** [comment]
im currently out of ideas.

do you have a legit apk build? from google play or from github releases..?
are othere devices (ios) on the same wifi network work fine?
is your problematic device working fine on _other_ wifi networks?

**Comment by sgornick** [comment]
> do you have a legit apk build? 

I had previously uninstalled and reinstalled, from Google Play, numerous times. Same behavior.  Just did so again, before composing this reply -- no difference.

> are othere devices (ios) on the same wifi network work fine?

Other mobile devices, I haven't been able to try.  I'll be getting a replacement mobile (also Android) in a couple weeks, I can try then on same wi-fi(s) but different Android, to see if that makes any difference..

> is your problematic device working fine on other wifi networks?

I finally connected to a different wifi to test against.  I get the same behavior on either wifi.  Even cycling (i.e., uninstalling and then installing) doesn't result in anything different -- i.e., still freezes as described above.

I did test now with v6.x now too:
Android version: 8.1.0, Android security patch level: October 5, 2020, Kernel version: 4.4.95
BlueWallet ver 6.0.2

**Comment by lestephane** [comment]
Same problem here (`Error: Could not fetch HD Bech32 balance` in self test) 
BlueWallet 6.0.3 Android: 7.0 security patch level October 5, 2017
electrum3.bluewallet.io:443

I'm surprised the phone can even start BlueWallet to be honest.

If you need more info (now that sgornick isn't the only one experiencing this), let me know.

I'm sending my IP to the email address mentioned with a link to this comment...

**Comment by sgornick** [comment]
Upon installing BlueWallet v6.0.6 Android, I no longer have this issue.

Closing as RESOLVED.  Thank you to whomever made the fix.

**Comment by sgornick** [comment]
I might have spoke too soon.   While the Create wallet succeeded without freezing, and even import had succeeded and I was able to see transactions, I don't think I had tried to run the Run Self-Test after that.

Nowadays I use BlueWallet on a halfway decent Android device (Samsung A10e, running Android 9), and I don't think I since have actually ran the Run Self-Test on either that older device (Alcatel 5041c, running Android 8.1.0) in which I reported this issue, nor on this Samsung Android device either.

Today, on my Samsung Android, I upgraded BlueWallet to v6.1.2. Because it is a new release, I wanted to try running the Self Test.  It worked, but then I found what is likely a regression so I reported that separately, as Issue #3193. 

I then wanted to see if that Self-Test behavior would occur with my old Alcatel Android.  Because this Alcatel Android has been factory reset since I closed this Issue back in March of this year,  I had to do a fresh install of BlueWallet.  First thing that surprised me is that the BlueWallet version shows v6.0.8, and not v6.1.2.   I then learned, after seeing Issue #3072, that this is occurring because of compatibility reasons (due to Tor), and v6.0.8 is still the latest BlueWallet release that is compatible with that device and Android operating system. OK, that's fine.

So then on that Alcatel Android, I tried to Run Self-Test.  But I am still getting the following error:

`Error: Could not fetch HD Bech32 balance`
 
So, I'm not sure if that Self-Test had ever worked successfully on that Alcatel Android device.   And at this point, that device is something I'm no longer going to be using.   So I will leave this Issue as Closed, but wanted to add this comment to clarify that while the freezing on Create Wallet issue as well as the failing Import issue have since been resolved for this Alcatel Android device, the Run Self-Test on it has not been resolved.

I don't plan to perform any further testing or troubleshooting on it, unless there's something you folks at BlueWallet wanted me to try.  Otherwise, this Alcatel Android device will be getting powered down permanently at some point.

So, thankfully I have the latest BlueWallet (v6.1.2) working well on my not-yet obsolete Samsung Android Device and am once again a happy BlueWallet user!
